XamarinCommunityToolkit/Effects/SafeAreaEffectRouter.ios.cs

```diff
@@ -24,24 +24,27 @@ protected override void OnAttached()
 
 			var insets = UIApplication.SharedApplication.Windows[0].SafeAreaInsets;
 
-			initialMargin = element.Margin;
-
 			if (insets.Top <= 0)
 				return;
 
+			initialMargin = element.Margin;
+
 			var safeArea = SafeAreaEffect.GetSafeArea(element);
 
```

Comment:
How about adding "isApplied" boolean filed instead.
We can set it to true in this line and then check if it's true in the **OnDeatached** method

Reply:
You could. But the entire point with conditional value types is that you don’t need two values for this.

Reply:
You could. But the entire reason for billable value types existing is exactly to avoid having two variables to accomplish this. Why not use all language features?